src/bgem/bspline/isec_surf_surf.py

```diff
@@ -88,589 +87,466 @@ def _main_curves(surf, axis):
+                    lpoint_list = []
                     for ipatch2 in intersectioned_patches2:
-                        iu2, iv2 = surf2.patch_id2pos(ipatch2)
+                        iu2, iv2 = other_surf.patch_id2pos(ipatch2)
+                        # TODO: rel_tol is not used, meaning should be clarified ...
                         uvt,  conv, xyz = curv_surf_isec.get_intersection(iu2, iv2, it, self.max_it,
                                                                             self.rel_tol, self.abs_tol)
                         if conv == 1:
-                            # Point A
+                            # Own Point
                             uv_a = np.zeros([2])
                             uv_a[axis] = w_val[curve_id]
                             uv_a[1 - axis] = uvt[2]
                             iuv_a = np.zeros([2], dtype=int)
                             iuv_a[axis] = patch[curve_id]
                             iuv_a[1 - axis] = it
-                            surf_point_a = SP.SurfacePoint(surf1, iuv_a, uv_a)
+                            own_point = SP.SurfacePoint(own_surf, iuv_a, uv_a)
 
-                            # Point B
+                            # Other Point
                             uv_b = uvt[0:2]
                             iuv_b = np.array([iu2, iv2])
-                            surf_point_b = SP.SurfacePoint(surf2, iuv_b, uv_b)
-
-                            point = IP.IsecPoint(surf_point_a, surf_point_b, xyz)
-
-                            point_list.append(point)
+                            other_point = SP.SurfacePoint(other_surf, iuv_b, uv_b)
 
-                            #if interval_intersections == 0:
-                            #    point_list.append(point)
-                            #    interval_intersections += 1
-                            #else:
-                            #    a = 1
-                                #check
+                            point = IP.IsecPoint(own_point, other_point, xyz)
 
 
-                            direction = surf_point_a.interface_flag[1-axis]
+                            # Check duplicities (cross points)
+                            direction = own_point.interface_flag[1 - axis]
                             if direction != 0:
                                 ind = [curve_id, curve_id]
                                 ind[1-axis] = it + int(0.5 * (direction + 1))
-                                crossing[tuple(ind)] = 1
-                            #break  # we consider only one point as an intersection of segment of a curve and a patch
-                            #check duplicities
-
+                                ind = tuple(ind)
+                                print(ind)
+                                if crossing[ind] == 1:
+                                    continue
+                                else:
+                                    crossing[ind] = 1
```

Comment:
Seems that we can replace the crossing indicator matrix. 
ind = curve_id, it + 'direction'
1. there will never be duplicate points with different curve_id
2. it + 'direction' can only match current it or it -1
So the detection of duplicities can be simplified using one or two bool flags.